GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/StorageClient.cs

```diff
@@ -62,5 +64,22 @@ public StorageClient(BaseClientService.Initializer initializer)
             };
             return new StorageClient(initializer);
         }
+
+        private static readonly Regex ValidBucketName = new Regex(@"^[0-9a-z.\-_]{1,222}$");
+
+        /// <summary>
+        /// Validates that a bucket only contains valid characters, and is not too long. This is far from
+        /// complete validation, but is all that's required to ensure that it's safe to include in a URL.
+        /// This method also checks for nullity, so callers don't need to do that first.
+        /// This method is internal rather than private for testing purposes.
+        /// </summary>
+        internal static void ValidateBucket(string bucket)
+        {
+            bucket.CheckNotNull(nameof(bucket));
+            if (!ValidBucketName.IsMatch(bucket))
+            {
+                throw new ArgumentException("Invalid bucket name - see https://cloud.google.com/storage/docs/bucket-naming", nameof(bucket));
```

Comment:
I think we should include the invalid value here, in case folks are seeing this exception in their logs.